src/NHSD.BuyingCatalogue.Identity.Api/Controllers/AccountController.cs

```diff
@@ -92,20 +89,37 @@ public async Task<IActionResult> Login(LoginViewModel viewModel)
             if (Url.IsLocalUrl(returnUrl))
                 return Redirect(returnUrl);
 
-            if (string.IsNullOrEmpty(returnUrl))
+            if (string.IsNullOrWhiteSpace(returnUrl))
                 return Redirect("~/");
 
-            // User might have clicked on a malicious link - should be logged
-            throw new Exception("Invalid return URL");
+            // Return URL could be malicious
+            throw new InvalidReturnUrlException();
         }
 
-        [HttpGet]
```

Comment:
The [Microsoft docs](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-3.1#exception-handler-page) state that error handler action methods should not be marked with HTTP method attributes as they can prevent some requests from reaching the method – this appears to be the case here.

Reply:
👍